sema/subyt/sources.py

```
@@ -350,6 +364,24 @@ def __exit__(self, *exc) -> None:
+class EmptySource(Source):
+    """
+    Fake Empty Source producing iterator over nothing.
+    """
+
+    def __init__(self) -> None:
+        super().__init__()
+
+    def __enter__(self) -> object:
+        return iter([])
+
+    def __exit__(self, *exc) -> None:
```

Ruff (0.8.2)

Missing type annotation for `*exc`

(ANN002)